### ISSUE TITLE
When Using PuTTY to Edit a Program in C++, NVDA Does Not Read the Letter Under the Cursor

### ISSUE THREAD

**Issue by LivingForGod1991** [nvaccess/nvda#6709]
This occurs with the latest Next snapshot of NVDA. When I amconnected to my school's Linux Lab with a program called PuTTY and editing a program, NVDA does not read the character under the cursor when I use the arrow keys to move around. It seems to be reading things that are a few characters away or that I already passed over. Sometimes, it even reads things from a few lines above where the cursor is. If I want to have some idea of where I am, I have to use the review cursor to read the area around the cursor.

**Comment by feerrenrut** [comment]
Could you clarify for me, is the behaviour better in a previous version of NVDA? If so, which version?
A work around for you may be to edit the files locally, and "synchronize" them with the remote directory. You can use a SCP (secure copy) client to do this, PSCP is a SCP client that comes with the Putty suite, and WinSCP is another SCP client. I hope this helps.

**Comment by leonardder** [comment]
I can confirm this issue, and it has been there for years, at least in my case. As far as I can say, it has to do with the latency involved with communication to the remote server. Strange enough, it can fix or improve the behaviour if you run the PuTTY session in full screen.

My theory is as follows:

1. You press an arrow key in PuTTY
2. Due to latency, it takes some time for PuTTY to update the local cursor to the remote position
3. Since NVDA doesn't know there is latency involved, it just instantly reports the position of the cursor, which is in fact the old position of the cursor as the cursor change isn't reflected visually yet.

Does this make sense?

**Comment by leonardder** [comment]
@Dkager: Do you know about this issue and is it just PuTTY related? How about TeraTerm, for example?

**Comment by dkager** [comment]
I don't recall seeing this. My connection has some latency, but usually under 30ms.

**Comment by jcsteh** [comment]
Duplicate of #4481. See https://github.com/nvaccess/nvda/issues/4481#issuecomment-155327931 for explanation.
